snuba/subscriptions/utils.py

```diff
@@ -0,0 +1,6 @@
+from enum import Enum
+
+
+class SchedulerMode(Enum):
+    IMMEDIATE = "immediate"
+    WAIT_FOR_SLOWEST_PARTITION = "wait-for-slowest"
```

Comment:
Let's rename these to 
```
class WatermarkMode:    # Or SchedulingWatermarkMode
       PARTITION = "partition"
       GLOBAL = "global"
```

This concept has a specific name in streaming system, using its own proper name will make the code easier to understand.